### DISCUSSION TITLE
Using ILI3941 display and XPT2046 Touchscreen

### DISCUSSION THREAD

**Discussion by JoergZ2** [micropython/micropython#9686] [Hardware & Peripherals]
I am using this board ([https://www.az-delivery.de/en/products/az-touch-wandgehauseset-mit-touchscreen-fur-esp8266-und-esp32]) and am looking for support for using the display (ili3941) and the touch panel (XPT2046); datasheet: [https://cdn.shopify.com/s/files/1/1509/1638/files/ArduiTouch_Wandgehauseset_mit_Touchscreen_fur_ESP8266_und_ESP32_Datenblatt.pdf?12457698508057470656]. Of all the scripts I have found so far, none have worked or the required libraries cannot be installed. Does anyone have experience with this board and can possibly provide me with a sample code. A "Hello World" would be enough for me. The main thing is that I see something on the display.

**Comment by robert-hh** [comment]
A driver for XPT2046 is here: https://github.com/robert-hh/XPT2046-touch-pad-driver
A driver for ILI3941 is here: https://github.com/mcauser/deshipu-micropython-ili9341
Besides that, https://awesome-micropython.com/ is always a good start for drivers and tools.

**Reply by JoergZ2** [reply]
Thank you very much for the tips. But unfortunately this is too little information (for me). I'm looking for a project/script that just puts anything on the screen. From there I could then reconstruct how the drivers are included and how graphical objects are constructed for rectangles, circles, lines, text, buttons, etc. I have only a little experience with Python but none (yet?) with microcontrollers or Micropython and would like to work out the topic. Therefore, as already described above, a "ready-made" script with a simple "Hello World" output or a simple quite banal touchscreen application would help me. At the moment I don't even know which GPIOS and SPI should be used, which of these connectors are connected to which sockets of the connector for the display, whether definitions for GPIO and/or SPI are made or adapted in the scripts. Therefore once again the question for or request for a working script, in which the drivers are already integrated. The drivers alone do not bring anything to the display. For this it needs a script that uses the classes and methods of the drivers.
